Explosive ducks now have a brief, dramatic delay before exploding

Rukofamicom opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Tripping a rubber ducky mine now paralyzes you while it plays its honk out before exploding. 

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
The dramatic effect makes the mine more enjoyable for the victim, and when you're getting blown up this can do a lot to lessen the buzzkill. Currently it's pretty common for the sound not to even play completely for the victim due to the deafening effect of the explosion that happens at the same time. 
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->

### This is both an example of my change working and also an example of the deafening effect preventing players from hearing the current sound correctly - this explosion DID make sound, just not sound that the lizard player would be able to hear. 
https://user-images.githubusercontent.com/9547572/168131287-cd0afdd9-7982-44b2-bd68-6c60f82fe811.mp4

I rearranged when the `triggered` variable is set to ensure that multiple people cannot activate the mine, duplicating its explosive effect or causing runtimes. 

## Changelog
:cl:
tweak: The exploding rubber ducky now has a dramatic delay before exploding so the victim can truly appreciate the humor of their situation. This delay does not allow the victim any opportunity to move away from the explosion.
fix: Mines now have to be smartmines to use smartmine logic. Explosive ducks (and other mines) are now 100% more dangerous for security personnel who were previously immune to funny. 
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Rukofamicom committed:
Update mines.dm

pavl1ko commented:
Watch me rush over there and throw the duck away, time dilation easily adds 10 more seconds to do this

Rukofamicom commented:
On the one hand I could correct exactly why that won't work. On the other I would love to see someone blow themselves up trying to attempt this so I think I won't this time.

Rukofamicom committed:
Now ignores stun immunity

The stun is not being used as a stun in the traditional sense, and this is not supposed to nerf mines.

Rukofamicom commented:
Now ignores stun immunity

PowerfulBacon commented on code/game/objects/effects/mines.dm:
```diff
@@ -116,8 +118,6 @@
 	to_chat(victim, "<span class='danger'>*click*</span>")
 
 /obj/effect/mine/proc/on_entered(datum/source, atom/movable/AM)
-	SIGNAL_HANDLER
```
You can't just remove signal handler from a signal handler. Use addtimer instead of sleep.

Rukofamicom replied:
fixed, and as an added bonus I also found and fixed a bug with smartmine detection.

PowerfulBacon reviewed (changes requested):
See above

Rukofamicom commented:
Will do, might be a couple days though

Rukofamicom committed:
Update mines.dm

Rukofamicom commented:
PR now additionally fixes a bug I don't think anyone noticed. 

```
/obj/effect/mine
	var/smartmine = 0
```

```
/obj/effect/mine/stun/smart
	smartmine = 1
```

smartmine is defined... and then never actually used in the smartmine check at all. 

```
if(ismob(AM))
	checksmartmine(AM)
	
/obj/effect/mine/proc/checksmartmine(mob/living/target)
	if(target)
		if(!target.has_mindshield_hud_icon())
			triggermine(target)
```

Every version of mine, including the rubber ducky traitor mines, have been ignoring mindshielded players.

PowerfulBacon commented on code/game/objects/effects/mines.dm:
```diff
@@ -118,26 +120,34 @@
 	if(ismob(AM))
 		checksmartmine(AM)
 	else
+		triggered = 1	//ensures multiple explosions aren't queued if/while the mine is delayed
```
```suggestion
		triggered = TRUE	//ensures multiple explosions aren't queued if/while the mine is delayed
```

PowerfulBacon commented on code/game/objects/effects/mines.dm:
```diff
@@ -118,26 +120,34 @@
+		if(smartmine && target.has_mindshield_hud_icon())
+			return
+		else if(dramatic_sound)
+			triggered = 1
```
```suggestion
			triggered = TRUE
```

Rukofamicom committed:
Update code/game/objects/effects/mines.dm

Co-authored-by: PowerfulBacon <26465327+PowerfulBacon@users.noreply.github.com>

Rukofamicom committed:
Update code/game/objects/effects/mines.dm

Co-authored-by: PowerfulBacon <26465327+PowerfulBacon@users.noreply.github.com>
